envoy/network/dns_resolver.h

```diff
@@ -26,6 +26,8 @@ class DnsResolverFactory : public Config::TypedFactory {
       const envoy::config::core::v3::TypedExtensionConfig& typed_dns_resolver_config) const PURE;
 
   std::string category() const override { return std::string(DnsResolverCategory); }
+  virtual void init() {}
```

Comment:
add doc to these methods, saying why they are needed.

Reply:
ACK